commit 3f31937e8ce0f152935f3dfd363549c060eafeea (refs/changes/32/3232/2)
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2018-01-19T15:45:33-07:00 (1 year, 9 months ago)
    
    MANTA-3417 long-running query probe does not work with PostgreSQL 9.6.3
    Reviewed by: David Pacheco <dap@joyent.com>

diff --git a/alarm_metadata/probe_templates/postgres.yaml b/alarm_metadata/probe_templates/postgres.yaml
index 323e5f3..9504296 100644
--- a/alarm_metadata/probe_templates/postgres.yaml
+++ b/alarm_metadata/probe_templates/postgres.yaml
@@ -243,7 +243,9 @@
         -
             type: cmd
             config:
-                cmd: "PATH=/opt/postgresql/current/bin:/opt/local/bin:$PATH psql -U postgres -c \"SELECT pid,usename,client_addr,client_port,waiting,state,NOW() - query_start AS elapsed, query FROM pg_stat_activity WHERE state != 'idle' AND query NOT LIKE 'autovacuum: %' AND NOW() - query_start > interval '120 seconds'\" | tail +3"
+                cmd: "if [ $(psql -V | cut -f 3 -d ' ' | cut -f 1,2 -d '.') = '9.6' ]; then psql -U postgres -c \"SELECT pid,usename,client_addr,client_port,wait_event,wait_event_type,state,NOW() - query_start AS elapsed, query FROM pg_stat_activity WHERE state != 'idle' AND query NOT LIKE 'autovacuum: %' AND NOW() - query_start > interval '120 seconds'\" | tail +3; else psql -U postgres -c \"SELECT pid,usename,client_addr,client_port,waiting,state,NOW() - query_start AS elapsed, query FROM pg_stat_activity WHERE state != 'idle' AND query NOT LIKE 'autovacuum: %' AND NOW() - query_start > interval '120 seconds'\" | tail +3; fi"
+                env:
+                  PATH: "/opt/postgresql/current/bin:/opt/local/bin:$PATH"
                 interval: 60
                 timeout: 30
                 threshold: 2
@@ -272,4 +274,3 @@
             If the query is being run by an administrator, it should probably be
             cancelled.  Otherwise, assess the health of the shard and determine
             why the query is taking so long to run.
-
